lib/app/shared/url_launcher.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+import 'package:url_launcher/url_launcher.dart';
+import 'package:mocktail/mocktail.dart';
+
+part 'url_launcher.g.dart';
+
+class MockUrlLauncher extends _$UrlLauncher with Mock implements UrlLauncher {}
+
+@riverpod
+class UrlLauncher extends _$UrlLauncher {
+  @override
+  FutureOr<void> build() => const AsyncData(null);
+
+  Future<void> launch(String? url) async {
+    if (url == null) return;
+
+    state = AsyncValue<void>.loading();
+
+    final Uri _url = Uri.parse(url);
+    state = await AsyncValue.guard(() async {
```

Comment:
look over riverpod.dev carefully, this does not need guard and i'm pretty sure does n0ot need to be a notifier at all, just a futureprovider.

fututeprovider automaticall includes the loading and error concepts, right?
your state is void, you're only wanting to find out error...